src/Components/InputField.jsx

```diff
@@ -0,0 +1,17 @@
+export default function InputField({ setup, state }) {
+  const { label, placeholder, type, required } = setup;
+  const [getter, setter] = state;
+
+  return (
+    <label className="input-field">
+      {label}
+      <input
+        type={type}
+        placeholder={placeholder}
+        value={getter}
+        onChange={(event) => setter(event.target.value)}
+        required={required}
+      />
+    </label>
+  );
+}
```

Comment:
A more standard approach to this would be:
```suggestion
export default function InputField({ setup, value, onChange }) {
  const { label, placeholder, type, required } = setup;

  return (
    <label className="input-field">
      {label}
      <input
        type={type}
        placeholder={placeholder}
        value={value}
        onChange={(event) => onChange(event.target.value)}
        required={required}
      />
    </label>
  );
}
```

The reason the above is standard React has to do with component optimization. By providing objects / arrays as component properties, we make it difficult for React to know if it should re-render the component or not. When using primitives or memoed functions React knows not to re-render the component. You can search for "what causes react to re-render" to get more info if you need it, or ping me :)